core/scripts/cli/sponsor/viewLiquidationDetails.js

```diff
@@ -1,24 +1,144 @@
+  // to receive rewards.
+  const ExpandedERC20 = artifacts.require("ExpandedERC20");
   const sponsorAddress = await getDefaultAccount(web3);
-  const display = `Liquidated at epoch time ${liquidation.liquidationTime} by ${liquidation.liquidator}`;
+  const liquidationTimeReadable = new Date(Number(liquidation.liquidationTime.toString()) * 1000);
+  const display = `Liquidated at time ${liquidationTimeReadable} by ${liquidation.liquidator}`;
   const backChoice = "Back";
-  const withdrawAction = "Withdraw";
+  // Sponsors can withdraw rewards only if a liquidation has been disputed successfully.
+  const withdrawAction = "Dispute succeeded; withdraw rewards";
   choices = [{ name: backChoice }];
+
   // Check if the sponsor can withdraw by seeing if `withdrawLiquidation` reverts.
   try {
+    // This will only succeed if the state is PENDING_DISPUTE and a price has resolved that will make the dispute succeed.
     await emp.withdrawLiquidation.call(id, sponsorAddress);
     choices.push({ name: withdrawAction });
   } catch (err) {
-    // Withdraw wouldn't work so it shouldn't be a valid option.
+    // Withdraw wouldn't work so it shouldn't be a valid option. Let's print out a detailed response to the user to indicate why
+    // `withdraw` will fail.
+
+    // The sponsor can only withdraw rewards from a successfully disputed liquidation.
+    if (liquidation.state === LiquidationStatesEnum.PRE_DISPUTE) {
+      // If liquidation state is PRE_DISPUTE and `withdrawLiquidation` fails, then the liquidation has either expired
+      // or is pre-expiry. Only a liquidator can withdraw from an expired liquidation.
+
+      const currentTime = web3.utils.toBN(await emp.getCurrentTime());
+      const liquidationExpirationTime = web3.utils
+        .toBN(liquidation.liquidationTime)
+        .add(web3.utils.toBN(await emp.liquidationLiveness()));
+      if (liquidationExpirationTime.lte(currentTime)) {
+        console.log("Cannot withdraw rewards from a liquidation that has expired");
+      } else {
+        console.log("Cannot withdraw rewards from a liquidation that is pre-expiry and has not been disputed");
+      }
+    } else if (liquidation.state === LiquidationStatesEnum.PENDING_DISPUTE) {
+      // If the liquidation state is PENDING_DISPUTE and `withdrawLiquidation` fails, then it indicates that either
+      // a price has not resolved yet, or a price has resolved that indicates that the dispute will FAIL.
+
+      // Fetch oracle price and use it to customize error message.
+      const Finder = artifacts.require("Finder");
+      let finder = await Finder.deployed();
+      const OracleInterface = artifacts.require("OracleInterface");
+      let oracle = await OracleInterface.at(
+        await finder.getImplementationAddress(web3.utils.utf8ToHex(interfaceName.Oracle))
+      );
+      if (await oracle.hasPrice(await emp.priceIdentifier(), liquidation.liquidationTime)) {
```

Comment:
nit: I think you'll need to add as the last argument to this and the `getPrice` call so they don't throw:
```
{ from: emp.address }
```

This is because these methods are only callable by registered contracts. Conveniently, for view functions, you're allowed to specify any address you want as the `from` address. If you don't do this, I think the function will throw. Because of some weirdness in web3 and some/all node implementations, throws in a view function seem to just resolve to `true` (this is what I saw in my console when testing). See here for more details on this weirdness: https://github.com/ethereum/solidity/issues/4840#issue-352539391.

Reply:
Just for reference, [here's](https://github.com/UMAprotocol/protocol/blob/29ad088075872e4d24c46cf2dd6dc59b09d3ad0b/core/contracts/oracle/implementation/Voting.sol#L254) the hasPrice() function with the `onlyRegisteredContracts()` modifier.

Reply:
Got it. Done and added a commend since this is esoteric